platform/iphone/app_delegate.mm

```diff
@@ -720,15 +187,50 @@ - (void)applicationWillTerminate:(UIApplication *)application {
 // notification panel by swiping from the upper part of the screen.
 
 - (void)applicationWillResignActive:(UIApplication *)application {
-	on_focus_out(view_controller, &is_focus_out);
+	[self toggleFocus:NO];
 }
 
 - (void)applicationDidBecomeActive:(UIApplication *)application {
-	on_focus_in(view_controller, &is_focus_out);
+	[self toggleFocus:YES];
+}
+
+- (void)toggleFocus:(BOOL)isFocused {
```

Comment:
I tried to combine what was previously two separate setters in to one more clear function. My thinking was that this would reduce errors in only calling one of the two methods required to cause the desired effects.